server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java

```diff
@@ -110,8 +111,27 @@ private ComparisonAuditController() {
   public static List<CastVoteRecord>
       getCVRsForSequenceNumbers(final County the_county,
                                 final List<Integer> the_seq_num_list) {
-    return BallotSelection.selectCVRs(the_seq_num_list,
-                                      the_county.id());
+    final List<CastVoteRecord> cvrs =
+        BallotSelection.selectCVRs(the_seq_num_list, the_county.id());
+
+    // A map of a CVR to a CVR so we can get a unique persisted entity from the
+    // database.
+    final Map<CastVoteRecord, CastVoteRecord> phantomCvrs =
+        cvrs.stream()
+            .filter(cvr -> PhantomBallots.isPhantomRecord(cvr))
+            .collect(Collectors.toMap(
+                Function.identity(),
+                Function.identity(),
+                (a, b) -> b));
+
+    // Assign database identifiers to newly-created phantom CVRs.
+    phantomCvrs.entrySet().stream()
```

Comment:
could you use `.distinct()` on this stream instead of the cvr to cvr map? I'm not fully understanding the purpose of the cvr to cvr map.

Reply:
The CVRs that come out of the BallotSelection are POJOs that the persistence layer doesn't know anything about. So, I have to just attach 1 of them to Hibernate so it doesn't try to insert multiple CVR rows. Since I want future newly-created CVRs that match (equality-wise) to *actually* become the single CVR object (identity-wise) that the persistence layer knows about, I make a map of CVR to "persisted CVR", so that when I walk the map again I can pull out that persisted CVR and use that downstream.

If I use `distinct()`, then I end up with a list that I have to walk to test membership (a bit less efficient) or a set which I can test for membership but I can't actually get at another object that is stored in it.

Reply:
If you see a cleaner way, though, let me know... I went through a couple iterations of this before I landed on something that worked and didn't frustratingly end up trying to save multiple phantom CVRs to the database.